File: python-location/domain/location_item.py
from datetime import timedelta

class LocationItem:
    def __init__(self, id, name):
        self.id = id
        self.name = name
        self.reservations = []

    def is_available(self, start_date, end_date):
        for res in self.reservations:
            if not (end_date < res['start'] or start_date > res['end']):
                return False
        return True

    def add_reservation(self, start_date, end_date):
        self.reservations.append({'start': start_date, 'end': end_date})

    def get_active_reservations(self, now):
        return [res for res in self.reservations if res['end'] >= now]

File: Javascript-location/domain/LocationItem.js
class LocationItem {
  constructor(id, name) {
    this.id = id;
    this.name = name;
    this.reservations = []; // La liste des périodes réservées 
  }

  isAvailable(startDate, durationDays) {
    const endDate = new Date(startDate);
    endDate.setDate(endDate.getDate() + durationDays);

    return !this.reservations.some(({ start, end }) => {
      return (
        (start <= endDate && end >= startDate)
      );
    });
  }

  addReservation(startDate, durationDays) {
    const endDate = new Date(startDate);
    endDate.setDate(endDate.getDate() + durationDays);

    this.reservations.push({ start: startDate, end: endDate });
  }
}

module.exports = LocationItem;